tests/v2/validation/provisioning/rke2/hostname_truncation_test.go

```diff
@@ -47,122 +48,38 @@ func (r *HostnameTruncationTestSuite) SetupSuite() {
 
 func (r *HostnameTruncationTestSuite) TestProvisioningRKE2Cluster() {
 	tests := []struct {
-		name                        string
-		machinePools                []provv1.RKEMachinePool
-		defaultHostnameLengthLimits []int
+		name                       string
+		poolLevelHostnameLength    []int
+		clusterLevelHostnameLength []int
 	}{
 		{
-			name: "Cluster level truncation",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name: namegen.RandStringLower(1),
-				},
-				{
-					Name: namegen.RandStringLower(31),
-				},
-				{
-					Name: namegen.RandStringLower(63),
-				},
-			},
-			defaultHostnameLengthLimits: []int{0, 10, 31, 63},
-		},
-		{
-			name: "Machine pool level truncation - 10 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 10,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 10,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 10,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 16, 63},
-		},
-		{
-			name: "Machine pool level truncation - 31 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 31,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
-		},
-		{
-			name: "Machine pool level truncation - 63 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 63,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 63,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 63,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
-		},
-		{
-			name: "Cluster and machine pool level truncation - 31 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name: namegen.RandStringLower(31),
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 31,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
+			name:                       "Cluster and Pool level truncation",
```

Comment:
I'm not a fan of removing all these test cases, we want to preserve the different test names so we know exactly what we're testing. Could explain more maybe I'm missing something.

Reply:
good point, I'll add a names field in for this.

Reply:
I'm still not quite seeing how we're keeping the same test cases. Maybe I missing something. We need all this reporting purposes

Reply:
Can you please explain why you remove all the test cases, I'm probably missing something so quick explanation can clear this up. Thanks!